src/runtime/slot-dom-consumer.ts

```diff
@@ -16,12 +16,13 @@ import {Arc} from './arc.js';
 import {SlotConnection} from './recipe/slot-connection.js';
 import {SlotConsumer, Content, Rendering} from './slot-consumer.js';
 import {ProvidedSlotContext} from './slot-context.js';
+import {Runtime} from './runtime.js';
 
 export interface DomRendering extends Rendering {
   liveDom?: Template;
 }
 
-const templateByName = new Map<string, HTMLElement>();
+const templateByName = () => Runtime.getRuntime().getCacheService().getOrCreateCache<string, HTMLElement>('templateByName');
```

Comment:
Might be useful to have a convenience `getOrCreateCache()` method directly on Runtime to avoid the get-get-get chaining.

Reply:
getRuntime() will be deprecated eventually, so there won't be too much chaining. We can take a look once we get the API more-or-less in shape.